src/Data/API/Value.hs

```diff
@@ -93,6 +95,7 @@ data Value = String  !T.Text
            | Bool    !Bool
            | Int     !Int
            | List    ![Value]
+           | Set     !(Set.Set Value)
```

Comment:
We need to make sure that the CBOR encoding/decoding of the "generic representation" `Value` is consistent with the specific CBOR encoding/decoding instances generated by TH for particular types. Here's a worry: what if the `Ord` instance for the specific type used an ordering that isn't consistent with the ordering of the corresponding `Value`s?

Maybe this is okay, provided the decoder doesn't assume that the values are sorted, because it can sort them when constructing the set in memory.